### PR TITLE
full system status on 2 LEDs

### PR DESCRIPTION
**Pull request by night-ghost**
+ buzzer pin as parameter and support for inverted buzzer

### CODE REVIEW

**Comment by tridge** [comment]
I'm fine with the BoardLed2 driver
For the buzzer, it may be better to get the full set of tones/music. Mark has just implemented that for HAL_ChibiOS based on what we already have for HAL_Linux. 
On the boards you want this for is the buzzer pin associated with a timer channel? Is the timer different from the ones used for PWM output on the board? If so we should be able to get the full ardupilot tones, startup tune etc. We'd just need to make the library that does that not be HAL specific.

**Comment by night-ghost** [comment]
>On the boards you want this for is the buzzer pin associated with a timer channel? Is the timer different from the ones used for PWM output on the board? 

this is board specific. Current implementation is intended for active buzzer wihich  has its own tone. Passive buzzer on timer's channel is on Cl-Racing F4 at least.

File: libraries/AP_Notify/AP_Notify.cpp [hunk 1]
```diff
@@ -33,6 +33,7 @@
 #include "DiscoLED.h"
 #include "Led_Sysfs.h"
 #include <stdio.h>
+ #include "AP_BoardLED2.h"
```
Comment (tridge) [inline libraries/AP_Notify/AP_Notify.cpp]: extra whitespace

File: libraries/AP_Notify/AP_Notify.cpp [hunk 2]
```diff
@@ -81,6 +83,10 @@ const AP_Param::GroupInfo AP_Notify::var_info[] = {
     // @User: Advanced
     AP_GROUPINFO("OREO_THEME", 4, AP_Notify, _oreo_theme, 0),
 
+#if !defined(BUZZER_PIN)
+    AP_GROUPINFO("BUZZ_PIN", 5, AP_Notify, _buzzer_pin, 0),
```
Comment (tridge) [inline libraries/AP_Notify/AP_Notify.cpp]: this needs param docs as comments

File: libraries/AP_Notify/Buzzer.cpp [hunk 3]
```diff
@@ -208,7 +217,210 @@ void Buzzer::on(bool turn_on)
...
+    _pattern = pattern_id;
+    _pattern_counter = 0;
+}
+
+#else
+
+// buzzer pin as parameter
+
+bool Buzzer::init()
+{
```
Comment (tridge) [inline libraries/AP_Notify/Buzzer.cpp]: why have two implementations of init() and update() for with and without BUZZER_PIN? If it is just setting the pin then just init a pin variable in one place and you are done.

File: libraries/AP_Notify/Buzzer.cpp [hunk 4]
```diff
@@ -208,7 +217,210 @@ void Buzzer::on(bool turn_on)
...
+    on(false);
+
+    // set initial boot states. This prevents us issuing a arming
+    // warning in plane and rover on every boot
+    _flags.armed = AP_Notify::flags.armed;
+    _flags.failsafe_battery = AP_Notify::flags.failsafe_battery;
+    return true;
+}
+
+// update - updates led according to timed_updated.  Should be called at 50Hz
```
Comment (tridge) [inline libraries/AP_Notify/Buzzer.cpp]: "update led" ?

File: libraries/AP_Notify/Buzzer.h [hunk 5]
```diff
@@ -19,14 +19,17 @@
 
 #include <AP_HAL/AP_HAL.h>
 
-#if CONFIG_HAL_BOARD == HAL_BOARD_VRBRAIN
+#ifndef BUZZER_PIN
+// better to define it in hal
+ #if CONFIG_HAL_BOARD == HAL_BOARD_VRBRAIN
  # define BUZZER_PIN    32
-#elif CONFIG_HAL_BOARD == HAL_BOARD_LINUX && CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BBBMINI
+ #elif CONFIG_HAL_BOARD == HAL_BOARD_LINUX && CONFIG_HAL_BOARD_SUBTYPE == HAL_BOARD_SUBTYPE_LINUX_BBBMINI
```
Comment (tridge) [inline libraries/AP_Notify/Buzzer.h]: please don't add in unrelated whitespace in your patches

Comment (night-ghost) [reply libraries/AP_Notify/Buzzer.h]: You don't like #if/#endif indents? It helps to see logical structure

Comment (night-ghost) [reply libraries/AP_Notify/Buzzer.h]: All anothers fixed

**Comment by magicrub** [comment]
Why did this get closed?

**Comment by night-ghost** [comment]
because requires too much time

**Comment by magicrub** [comment]
@night-ghost you're not the only one spending time on your pull requests.

**Comment by tridge** [comment]
I've forced pushed a small change to remove some commented out code and rebased on master.
Looks good to me now, ready to merge when passed CI.
Thanks for fixing the issues @night-ghost
